### PR TITLE
Verify GPG signatures on Python source tarballs

### PR DESCRIPTION
**Pull request by trishankatdatadog**
Verify GPG signatures on Python source tarballs before building and installing them.

Includes a copy of public keys downloaded from Python.org over SSL/TLS. Please double-check.

### CODE REVIEW

**Comment by trishankatdatadog** [comment]
Probably want to put `gpg` in `PYTHON_COMPILE_DEPS`, so that . it's removed at the end of build.

**Comment by ogrisel** [comment]
LGTM. Thanks for working on improving the security of the build pipeline.

File: docker/build_scripts/build.sh [hunk 1]
```diff
@@ -96,6 +97,8 @@ rm -rf sqlite-autoconf-3160200*
 # statically. We delete openssl afterwards.)
 build_openssl $OPENSSL_ROOT $OPENSSL_HASH
 mkdir -p /opt/python
+# Import public keys used to verify these signatures.
+gpg --import $MY_DIR/cpython-pubkeys.txt
```
Comment (ogrisel) [inline docker/build_scripts/build.sh]: Actually please add the source URL (where those public keys where downloaded from) to the inline comment.

Comment (ogrisel) [reply docker/build_scripts/build.sh]: I believe it's https://www.python.org/static/files/pubkeys.txt

**Comment by trishankatdatadog** [comment]
My pleasure to improve the pipeline security, and done!

**Comment by ogrisel** [comment]
Merged!

**Pull request merged**